service/topology_coordinator.cc

```diff
@@ -1558,6 +1560,12 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
                 co_return true;
             }
 
+            if (auto [did_migrate, guard_opt] = co_await maybe_migrate_system_tables(std::move(guard)); did_migrate) {
```

Comment:
Why do you need both `did_migrate` and `guard_opt` if `guard_opt` is engaged only when `did_migrate == true`

Reply:
technically I don't need `did_migrate` but I thought it made more sense from logical point of view to have this condition than checking if `guard_opt` is engaged

Reply:
i do not see why. `std::optional` already is a `bool` as well. If guard is returned it means it was not consumed so there was no migration otherwise there was.

Reply:
ok, done